native/Avalonia.Native/src/OSX/WindowOverlayImpl.mm

```diff
@@ -137,10 +138,12 @@
             auto key = VirtualKeyFromScanCode(scanCode, [event modifierFlags]);
             
             uint64_t timestamp = static_cast<uint64_t>([event timestamp] * 1000);
-            AvnInputModifiers modifiers = Windows; // Windows is equivalent to CMD
+            AvnInputModifiers modifiers = GetCommandModifier([event modifierFlags]); 
             AvnRawKeyEventType type;
-            
-            if ([event type] == NSEventTypeKeyDown)
+
+            // Type flag change with the set modifier is a key down. 
+            // Same with the unset modifier is a key up. [When the modifier key is released, the flag changes to 0x0]
+            if (modifiers != AvnInputModifiersNone)
```

Comment:
Changing from `([event type] == NSEventTypeKeyDown)` to `(modifiers != AvnInputModifiersNone)` does not look right.

This might work for the modifier keys, but remember that this monitor is also for key combinations.

Like:
Meta pressed
C pressed
C released
Meta released

Reply:
I tested for the key combination of Command+C / Command+V / Command+Z / Command+X and Shift+ combinations.
Did not observe an issue. 

Initially I used the expression ([event type] == NSEventTypeKeyDown) ||  (modifiers != AvnInputModifiersNone), as it did not make an impact during the developer testing, I removed it.

I will check more. If you have suspectable combinations, please let me know.

Reply:
Thanks for the input. I have modified to include all key combinations now, not only looking at the Modifier events solely.

# How Events are Managed

## When Modifier+Key

WOI: Dispatching Key Flags =1048576, Event=12
WOI: Captured Key Event Flags =1048576, Event=12
WOI: Dispatching Key Flags =1048576, Event=10
WOI: Captured Key Event Flags =1048576, Event=10
WOI: Dispatching Key Flags =1048576, Event=11
WOI: Captured Key Event Flags =1048576, Event=11
WOI: Dispatching Key Flags =0, Event=12
WOI: Captured Key Event Flags =0, Event=12


## When Modifier Only (Needed for mouse events)

WOI: Dispatching Key Flags =1048576, Event=12
WOI: Captured Key Event Flags =1048576, Event=12
WOI: Dispatching Key Flags =0, Event=12
WOI: Captured Key Event Flags =0, Event=12

## Any other key (Which are filtered out)

WOI: Dispatching Key Flags =0, Event=10
WOI: Dispatching Key Flags =0, Event=11